Source/WebCore/css/parser/CSSPropertyParserConsumer+Image.cpp

```diff
@@ -0,0 +1,1185 @@
+        if (auto horizontal = consumeIdentUsingMapping(range, horizontalMap))
+            return std::make_pair(*horizontal, knownVertical);
+        return knownVertical;
+    };
+
+    auto consumeKeywordGradientLine = [&](CSSParserTokenRange& range) -> std::optional<CSSLinearGradientValue::GradientLine> {
+        ASSERT(range.peek().id() == CSSValueTo);
+        range.consumeIncludingWhitespace();
+
+        switch (range.peek().id()) {
+        case CSSValueLeft:
+            range.consumeIncludingWhitespace();
+            return consumeKeywordGradientLineKnownHorizontal(range, CSSLinearGradientValue::Horizontal::Left);
+        case CSSValueRight:
+            range.consumeIncludingWhitespace();
+            return consumeKeywordGradientLineKnownHorizontal(range, CSSLinearGradientValue::Horizontal::Right);
+        case CSSValueTop:
+            range.consumeIncludingWhitespace();
+            return consumeKeywordGradientLineKnownVertical(range, CSSLinearGradientValue::Vertical::Top);
+        case CSSValueBottom:
+            range.consumeIncludingWhitespace();
+            return consumeKeywordGradientLineKnownVertical(range, CSSLinearGradientValue::Vertical::Bottom);
+        default:
+            return { };
+        }
+    };
+
+    std::optional<ColorInterpolationMethod> colorInterpolationMethod;
+
+    if (range.peek().id() == CSSValueIn) {
+        colorInterpolationMethod = consumeColorInterpolationMethod(range);
+        if (!colorInterpolationMethod)
+            return nullptr;
+    }
+
+    std::optional<CSSLinearGradientValue::GradientLine> gradientLine;
+
+    const auto angleConsumeOptions = CSSPropertyParserOptions {
+        .parserMode = context.mode,
+        .unitless = UnitlessQuirk::Forbid,
+        .unitlessZero = UnitlessZeroQuirk::Allow
+    };
+
+    if (auto angle = MetaConsumer<AngleRaw>::consume(range, { }, angleConsumeOptions))
+        gradientLine = WTF::switchOn(WTFMove(*angle), [](auto&& value) -> CSSLinearGradientValue::GradientLine { return value; });
+    else if (range.peek().id() == CSSValueTo) {
+        auto keywordGradientLine = consumeKeywordGradientLine(range);
+        if (!keywordGradientLine)
+            return nullptr;
+        gradientLine = WTFMove(*keywordGradientLine);
+    }
+
+    if (gradientLine && !colorInterpolationMethod && range.peek().id() == CSSValueIn) {
+        colorInterpolationMethod = consumeColorInterpolationMethod(range);
+        if (!colorInterpolationMethod)
+            return nullptr;
+    }
+
+    if (gradientLine || colorInterpolationMethod) {
+        if (!consumeCommaIncludingWhitespace(range))
+            return nullptr;
+    }
+
+    auto stops = consumeLengthColorStopList(range, context, SupportsColorHints::Yes);
+    if (!stops)
+        return nullptr;
+
+    auto computedColorInterpolationMethod = computeGradientColorInterpolationMethod(colorInterpolationMethod, *stops);
```

Comment:
Not sure we need the local variable for this.

Reply:
I think we do. `stops` is WTFMove()'d in the function call, and since the order of evaluation of function call arguments is not defined, stops might be moved before this would have been called.